src/module.c

```diff
@@ -190,7 +190,7 @@ static jl_binding_t *jl_get_binding_(jl_module_t *m, jl_sym_t *var, modstack_t *
                 if (tempb == NULL || tempb->owner == NULL)
                     // couldn't resolve; try next using (see issue #6105)
                     continue;
-                if (owner != NULL && tempb->owner != b->owner &&
+                if (owner != NULL && tempb->owner != b->owner && !tempb->deprecated &&
```

Comment:
It looks like this will not work if we happen to find a deprecated binding first.